### PR TITLE
docs: Update README with minimum gradle version

### PR DESCRIPTION
**Pull request by jjliu15**
- Adds minimum gradle plugin version to the README
- Also fixes formatting issues and adds links to the codelab

## Related Issues
https://github.com/googleads/googleads-mobile-flutter/issues/20
https://github.com/googleads/googleads-mobile-flutter/issues/26

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ x] All existing and new tests are passing.
- [ x] I updated/added relevant documentation (doc comments with `///`).
- [ x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [ x] I read and followed the [Flutter Style Guide].
- [ x] I signed the [CLA].
- [ x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x ] No, this is *not* a breaking change.

### CODE REVIEW

**Review by bparrishMines — COMMENTED** [review]
Could you also update the gradle version of the example app so that we verify the version works.

File: README.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Note: This plugin also contains support for **[Google Ad Manager](https://admana
     *   Android Studio 3.2 or higher
     *   Target Android API level 19 or higher
     *   Set `compileSdkVersion` to 28 or higher
+    *   Android Gradle 4.1 or higher (this is the version supported by Flutter out of the box)
```
Comment (blasten) [inline README.md:19]:
```suggestion
    *   Android Gradle Plugin 4.1 or higher (this is the version supported by Flutter 2.0 out of the box)
```

**Comment by blasten** [comment]
@jjliu15  I noted this while looking at the readme:

The main heading of the readme should be something like "Google Mobile Ads for Flutter". Every other heading should have an extra `#` in the heading.

**Review by blasten — APPROVED** [review]
LGTM

**Pull request merged**